autogrow-PFS-quota/ARMtemplate.json

```diff
@@ -85,147 +85,151 @@
-        ],
-        "properties": {
-          "roleDefinitionId": "[resourceId('Microsoft.Authorization/roleDefinitions', parameters('roleDefinitionId'))]",
-          "principalId": "[reference(resourceId('Microsoft.Web/sites', variables('functionAppName')),'2019-08-01', 'full').identity.principalId]",
-          "scope": "[resourceGroup().id]"
+    {
+      "apiVersion": "2019-08-01",
+      "name": "[concat(variables('functionAppName'),'/', parameters('functionName'))]",
+      "type": "Microsoft.Web/sites/functions",
+      "dependsOn": [
+        "[resourceId('Microsoft.Web/sites', variables('functionAppName'))]"
+      ],
+      "properties": {
+        "config": {
+          "bindings": [
+            {
+              "name": "Timer",
+              "type": "timerTrigger",
+              "direction": "in",
+              "schedule": "0 */5 * * * *"
+            }
+          ]
+        },
+        "files": {
+          "azure-functions-script.ps1": "# Input bindings are passed in via param block.\r\nparam($Timer)\r\n\r\n# Get the current universal time in the default string format.\r\n$currentUTCtime = (Get-Date).ToUniversalTime()\r\n\r\n# The 'IsPastDue' property is 'true' when the current function invocation is later than scheduled.\r\nif ($Timer.IsPastDue) {\r\n    Write-Host \"PowerShell timer is running late!\"\r\n}\r\n\r\n#Variable Definitions\r\n$environment = $env:environment\r\n$subscription_id = $env:targetSubscriptionId\r\n$resource_group = $env:targetResourceGroup\r\n$storage_account_name = $env:targetStorageAccountName\r\n$file_share_name = $env:targetFileShareName\r\n\r\n#Connect to Azure and Import Az Module\r\nImport-Module -Name Az.Accounts\r\nImport-Module -Name Az.Storage\r\nConnect-AzAccount -Environment $environment -Subscription $subscription_id -Identity\r\n\r\n# Get file share\r\n$PFS = Get-AzRmStorageShare -ResourceGroupName $resource_group -StorageAccountName $storage_account_name -Name $file_share_name -GetShareUsage\r\n\r\n# Get provisioned capacity and used capacity\r\n$ProvisionedCapacity = $PFS.QuotaGiB\r\n$UsedCapacity = $PFS.ShareUsageBytes\r\nWrite-Host \"Provisioned Capacity:\" $ProvisionedCapacity\r\nWrite-Host \"Share Usage Bytes:\" $UsedCapacity\r\n\r\n# get storage account\r\n$StorageAccount = Get-AzStorageAccount -ResourceGroupName $resource_group -AccountName $storage_account_name\r\n\r\n# if less than 20% of provisioned capacity is remaining, increase provisioned capacity by 20%\r\nif (($ProvisionedCapacity - ($UsedCapacity / ([Math]::Pow(2,30)))) -lt ($ProvisionedCapacity*0.2)) {\r\n     $Quota = $ProvisionedCapacity*1.2\r\n     Update-AzRmStorageShare -StorageAccount $StorageAccount -Name $file_share_name -QuotaGiB $Quota\r\n     $ProvisionedCapacity = $Quota\r\nWrite-Host \"New Provisioned Capacity:\" $ProvisionedCapacity\r\n}\r\n\r\n# Write an information log with the current time.\r\nWrite-Host \"PowerShell timer trigger function ran! TIME:\" $currentUTCtime\r\n"
```

Comment:
Is there an easier way to look at the diff of the powershell script? Currently the entire script is showing as a single line making is difficult to review the actual changes.

Reply:
Future reference, you could save the line of code as a JSON object and save the file to your file system.  Then run the following code: 
$JSON = Get-Content -Path ".\<fileName>.json"
$Script = ($JSON | ConvertFrom-Json)."azure-functions-script.ps1"

However, here's the code:
# Input bindings are passed in via param block.
param($Timer)

# Get the current universal time in the default string format.
$currentUTCtime = (Get-Date).ToUniversalTime()

# The 'IsPastDue' property is 'true' when the current function invocation is later than scheduled.
if ($Timer.IsPastDue) {
    Write-Host "PowerShell timer is running late!"
}

#Variable Definitions
$environment = $env:environment
$subscription_id = $env:targetSubscriptionId
$resource_group = $env:targetResourceGroup
$storage_account_name = $env:targetStorageAccountName
$file_share_name = $env:targetFileShareName

#Connect to Azure and Import Az Module
Import-Module -Name Az.Accounts
Import-Module -Name Az.Storage
Connect-AzAccount -Environment $environment -Subscription $subscription_id -Identity

# Get file share
$PFS = Get-AzRmStorageShare -ResourceGroupName $resource_group -StorageAccountName $storage_account_name -Name $file_share_name -GetShareUsage

# Get provisioned capacity and used capacity
$ProvisionedCapacity = $PFS.QuotaGiB
$UsedCapacity = $PFS.ShareUsageBytes
Write-Host "Provisioned Capacity:" $ProvisionedCapacity
Write-Host "Share Usage Bytes:" $UsedCapacity

# get storage account
$StorageAccount = Get-AzStorageAccount -ResourceGroupName $resource_group -AccountName $storage_account_name

# if less than 20% of provisioned capacity is remaining, increase provisioned capacity by 20%
if (($ProvisionedCapacity - ($UsedCapacity / ([Math]::Pow(2,30)))) -lt ($ProvisionedCapacity*0.2)) {   
     $Quota = $ProvisionedCapacity*1.2
     Update-AzRmStorageShare -StorageAccount $StorageAccount -Name $file_share_name -QuotaGiB $Quota   
     $ProvisionedCapacity = $Quota
Write-Host "New Provisioned Capacity:" $ProvisionedCapacity
}

# Write an information log with the current time.
Write-Host "PowerShell timer trigger function ran! TIME:" $currentUTCtime